src/main/java/org/opensearch/knn/index/codec/jvector/JVectorCompoundFormat.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.jvector;
+
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.codecs.CompoundDirectory;
+import org.apache.lucene.codecs.CompoundFormat;
+import org.apache.lucene.index.CorruptIndexException;
+import org.apache.lucene.index.IndexFileNames;
+import org.apache.lucene.index.SegmentInfo;
+import org.apache.lucene.store.*;
+import org.apache.lucene.util.CollectionUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+// TODO: This needs to be moved under the same package name as the Lucene internal package name for {@link Lucene90CompoundReader}
+// this way the internal package constants can be accessed directly and we can avoid duplicating them.
+@Log4j2
+public class JVectorCompoundFormat extends CompoundFormat {
```

Comment:
Can you remind me if this is still needed after our discuss? I remember that @navneet1v mentioned on slack that we dont need this. It seems like this is the only reason we'd need to create a custom codec for jvector as opposed to just the knnvectorsformat. So, Id like to get rid of it if possible.

Reply:
It can potentially be removed after `RandomAccessReader` for jVector is delegated to `IndexInput`.